content/en/agent/proxy.md

```diff
@@ -142,17 +142,25 @@ Do not forget to [restart the Agent][1] for the new settings to take effect.
 
 ## HAProxy
 
-[HAProxy][1] is a free, fast, and reliable solution offering proxying for TCP and HTTP applications. While HAProxy is usually used as a load balancer to distribute incoming requests to pools servers, you can also use it to proxy Agent traffic to Datadog from hosts that have no outside connectivity.
-
-This is the best option if you do not have a web proxy readily available in your network, and you wish to proxy a large number of Agents. In some cases, a single HAProxy instance is sufficient to handle local Agent traffic in your network-each proxy can accommodate upwards of 1000 Agents. **Note**: This figure is a conservative estimate based on the performance of m3.xl instances specifically. Numerous network-related variables can influence load on proxies. As always, deploy under a watchful eye. See the [HAProxy documentation][2] for additional information.
+[HAProxy][1] is a free, fast, and reliable solution offering proxying for TCP and HTTP applications. While HAProxy is usually used as a load balancer to distribute incoming requests to pools servers, you can also use it to proxy Agent traffic to Datadog from hosts that have no outside connectivity :
 
 `agent -> haproxy -> Datadog`
 
+This is the best option if you do not have a web proxy readily available in your network, and you wish to proxy a large number of Agents. In some cases, a single HAProxy instance is sufficient to handle local Agent traffic in your network as each proxy can accommodate upwards connections of 1000 Agents. **Note**: This figure is a conservative estimate based on the performance of m3.xl instances specifically. Numerous network-related variables can influence load on proxies. As always, deploy under a watchful eye. See the [HAProxy documentation][2] for additional information.
+
+The communication between HAProxy and Datadog will always be encrypted and it is recommended to encrypt the communication between the Agent host and the HAProxy host if they are not part of the same local network.
+In order to encrypt data between the Agent and HAProxy, you will need to create a Subject Alternative Name (SAN) certificate for the HAProxy host. This certificate should contain both the public certificate and private key. See this [HAProxy blog post][3] for more information.
```

Comment:
> Isn't that just a regular SSL/TLS certificate? Do we mean an x509 certificate?

Yeah, we mean an x509 certificate here but if I correctly understand, there is a field named `Common Name` (or CN). which is deprecated by the [RFC 2818](https://datatracker.ietf.org/doc/html/rfc2818#section-3.1) in 2000 and I believe you can still create certificate using `CN` instead of `SAN`.

```
If a subjectAltName extension of type dNSName is present, that MUST be used as the identity. 
...
Although the use of the Common Name is existing practice, it is deprecated 
and Certification Authorities are encouraged to use the dNSName instead
```

 I added this precision because the Agent throws an error when trying to verify a certificate with CN and asks for SAN instead. Maybe should I say `you will need to create a x509 certificate with Subject Alternative Name (SAN) extension` ?

---

> I think this can be reworded just to say that "you will need to create a Subject Alternative Name (SAN) certificate and its private key for the HAProxy host" ? 

The problem here is that NGINX has 2 attributes : 

-  `ssl_certificate`  for the certificate (usually `.crt`)
-  `ssl_certificate_key` for the key (usually `.key`)

But HAProxy has only one attribute `ssl crt <cert.pem>` where `<cert.pem>` is a file containing both the certificate and the private key (can be created with `cat certificate.crt private_key.key > cert.pem`). I wanted to emphasise the fact that the certificate and private key must be in the same file but I agree that it is unclear so feel free to give any suggestions on how to rephrase it 🙏

Reply:
> Maybe should I say `you will need to create a x509 certificate with Subject Alternative Name (SAN) extension`

You can't have a SAN field without it being an x509 certificate I believe, this works great - makes it clear about what they need to do. 

> feel free to give any suggestions on how to rephrase it

Maybe we can call it a certificate bundle? WDYT?
```
This certificate bundle (*.pem) should contain both the public certificate and private key. See this [HAProxy blog post][3] for more information.
```

Reply:
Done here : cca6a1513774cf4d69a0b2d28a621cbba772027e